### PR TITLE
[Home][ArtworkRail] Header Visuals

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] New component for header
- [x] Related Artist state (based on + follow button)

simulator screen shot aug 16 2016 10 46 20 pm

- [x] Additional content state

simulator screen shot aug 16 2016 10 49 54 pm

- [x] No additional content state
- [x] Polish
- [x] iPad styling

rail

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Hello, world.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by orta** [comment]
Looks good so far 👍 

I'm going to start introducing a testing culture in this repo very soon, would recommend exploring jest-snapshots to start adding tests to cover your new component 💃

File: lib/components/home/artwork_rails/artwork_rail_header.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      <View style={styles.container}>
+        { this.props.rail.context && this.followAnnotation() }
+        <SerifText numberOfLines={2} style={styles.title}>{ this.props.rail.title }</SerifText>
+        { this.props.rail.context && this.actionButton() }
+      </View>
+    )
+  }
+
+  followAnnotation() {
+    // string comparing feels super weird but making a javascript 'enum' might be worse?
```
Comment (sarahscott) [inline lib/components/home/artwork_rails/artwork_rail_header.js]: Feedback & suggestions here in particular would be appreciated 💙 .

Comment (orta) [inline lib/components/home/artwork_rails/artwork_rail_header.js]: Good Q, Flow does [support them](https://flowtype.org/docs/builtins.html#literal-types) but if you only have one option, I'm not sure it makes sense to do overall.

 the `__typename` does feel awkward though, is this something that occurs from graphql?

Comment (sarahscott) [inline lib/components/home/artwork_rails/artwork_rail_header.js]: Yeah, [this](http://graphql.org/docs/queries/) explains the `__typename` thing. [Force](http://graphql.org/docs/queries/) handles each `type` with its own query, but that's not necessary for us at this time.

Comment (orta) [inline lib/components/home/artwork_rails/artwork_rail_header.js]: This makes total sense - then for now, I think you are good with the `if` - if there was more than one, then the switch is the way to go

Comment (orta) [inline lib/components/home/artwork_rails/artwork_rail_header.js]: an alternative would be to `const showSubtitle = this.props.rail.context.__typename === 'HomePageModuleContextRelatedArtist'` it out, and force some specify, but I'm not sure you gain too much

**Comment by sarahscott** [comment]
cc @katarinabatina check out the gif 💫

**Comment by sarahscott** [comment]
I'm happy with this as a standalone PR and I'll make touch-handling into its own tomorrow.

**Comment by orta** [comment]
Cool, last PR I'm going to start merging without tests though #241 will bring it in probably tomorrow

**Comment by orta** [comment]
💃

**Comment by katarinabatina** [comment]
holy crapoly this is lookin pretty sweet @sarahscott !

**Comment by orta** [comment]
^ big time 👍

**Comment by katarinabatina** [comment]
ship a betaaaaaa

**Comment by orta** [comment]
My TODO first thing tomorrow now

**Pull request merged**